### PR TITLE
[WiP] Plasma 5.21

### PR DESCRIPTION
**Pull request by Johnnynator**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

File: srcpkgs/kwin/patches/include-fcntl.patch [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ #include <stdio.h>
+ #include <stdlib.h>
+ #include <string.h>
+--- helpers/wayland_wrapper/wl-socket.c	2021-01-23 23:26:30.050347671 +0100
++++ -	2021-01-23 23:30:37.754016254 +0100
+@@ -17,6 +17,7 @@
+ #include <sys/file.h>
+ #include <sys/socket.h>
+ #include <sys/stat.h>
++#define _GNU_SOURCE
```
Comment (ericonr) [inline srcpkgs/kwin/patches/include-fcntl.patch:17]: Ideally this define should go before any include to avoid weird issues.

**Review by ericonr — COMMENTED** [review]
Maybe fix `KDE_SITE` so we don't end up requiring all of those substitutions? Not sure which one is simpler, especially now that you've done most of the work.

**Comment by Johnnynator** [comment]
> Maybe fix `KDE_SITE` so we don't end up requiring all of those substitutions? Not sure which one is simpler, especially now that you've done most of the work.

Srtictly speaking, we do not require these substitutions ever, since we do not ship unstable software.

**Comment by toluschr** [comment]
Missing dependency `libSM-devel` for kscreenlocker

**Pull request closed without merging**